### ISSUE TITLE
Missing factory for StopWorkers

### ISSUE THREAD

**Issue by pimjansen** [netglue/laminas-messenger#196]
Hey,

I noticed that the `messenger:stop-workers` command is not implemented. This is not a problem however i am struggling a but to implement the factory myself where the command requires a `CacheItemPoolInterface`

Any guide on how to handle this? It would be a good addition to add the factory to this lib i suppose.

Thanks!

**Comment by gsteel** [comment]
Good idea to implement a factory here 👍

The cache pool will need to be identified in configuration, for example:

```php
return [
  'symfony' => [
    'messenger' => [
      'signalCachePool' => 'MyCachePoolServiceId',
    ],
  ],
  'dependencies' => [
    'factories' => [
      'MyCachePoolServiceId' => SomeFactory::class,
    ],
  ]
];
```

With the above, a factory for the command would then look up the service id in `symfony.messenger.signalCachePool` and fetch that from the container.

In order for the command to do anything, the listener `StopWorkerOnRestartSignalListener` has to be attached to each worker - this listener also needs the same cache pool.

The main problem for this lib is that the Stop command will need to be optional because its factory will throw an exception when there's no cache pool configured - we can't provide some kind of default cache because the cache needs to be distributed in order for it to work between processes.

**Comment by pimjansen** [comment]
Yeah thats why im reaching out since it important to see what this lib wants.

We can ship a default factory that resolves to a factory on a static key? From there everyone can use it and resolve to their own implementation.

What do you think?
